src/amo/components/UserProfileEdit/index.js

```diff
@@ -173,6 +178,46 @@ export class UserProfileEditBase extends React.Component<Props, State> {
     }
   }
 
+  onDeleteProfile = (e: SyntheticEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    this.setState({ showProfileDeletionModal: true });
+  }
+
+  onCancelProfileDeletion = (e: SyntheticEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    this.setState({ showProfileDeletionModal: false });
+  }
+
+  onConfirmProfileDeletion = (e: SyntheticEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    const {
+      clientApp,
+      currentUser,
+      dispatch,
+      errorHandler,
+      lang,
+      router,
+      user,
+    } = this.props;
+
+    invariant(currentUser, 'currentUser is required');
+    invariant(user, 'user is required');
+
+    dispatch(deleteUserAccount({
+      errorHandlerId: errorHandler.id,
+      userId: user.id,
+    }));
+
+    if (currentUser.id === user.id) {
+      dispatch(logOutUser());
+    }
+
+    router.push(`/${lang}/${clientApp}`);
```

Comment:
Would it be nice to have a route that we could push that just says "display the home page", as opposed to having to construct one using `lang` and `clientApp`? It just seems like that shouldn't be necessary.

Reply:
I don't think we could do better for now, but we should definitely investigate once we upgrade React Router or move to something else (https://github.com/mozilla/addons-frontend/issues/2161) .